src/univers/version_range.py

```diff
@@ -245,35 +279,66 @@ def from_native(cls, string):
-        anyof_constraints = []
-        if isinstance(clause, AnyOf):
-            for allof_clause in clause.clauses:
-                anyof_constraints.extend(get_allof_constraints(cls, allof_clause))
-        elif isinstance(clause, AllOf):
-            alloc = get_allof_constraints(cls, clause)
-            anyof_constraints.extend(alloc)
-        else:
-            raise ValueError(f"Unknown clause type: {spec!r}")
-
-        return cls(constraints=anyof_constraints)
 
+        if string == "*":
+            return cls(
+                constraints=[
+                    VersionConstraint.from_string(string="*", version_class=cls.version_class)
+                ]
+            )
 
-def get_allof_constraints(cls, clause):
-    """
-    Return a list of VersionConstraint given an AllOf ``clause``.
-    """
-    assert isinstance(clause, AllOf)
-    allof_constraints = []
-    for constraint in clause.clauses:
-        comparator = cls.vers_by_native_comparators[constraint.operator]
-        version = cls.version_class(str(constraint.target))
-        constraint = VersionConstraint(comparator=comparator, version=version)
-        allof_constraints.append(constraint)
-    return allof_constraints
+        constraints = []
+        vrc = cls.version_class
+        # A constraint item can be a comparator or a version or a version with comparator
+        # If it's empty continue
+        # If it's in `vers_by_native_comparators`, append it with the comparator and continue
+        # If it's a version, make version constraint from the version and use the comparator from the previous item and make comparator empty
+        # If it's a version with comparator, use split_req to get version and comparator to form constraint and make comparator empty
+
+        for range in string.split("||"):
+            if " - " in range:
+                constraints.extend(get_npm_version_constraints_from_semver_npm_spec(range, cls))
+                continue
+            comparator = ""
+            for constraint in range.split(" "):
+                if not constraint:
+                    continue
+                if "".join([comparator, constraint]) in cls.vers_by_native_comparators:
+                    comparator = "".join([comparator, constraint])
+                    comparator = cls.vers_by_native_comparators[comparator]
+                    continue
+                if comparator:
+                    if constraint.endswith(".x"):
+                        constraints.extend(
+                            get_npm_version_constraints_from_semver_npm_spec(constraint, cls)
+                        )
+                        continue
+                    constraint = constraint.lstrip("vV")
+                    constraints.append(
+                        VersionConstraint(comparator=comparator, version=vrc(constraint))
+                    )
+                else:
+                    if constraint.endswith(".x"):
```

Comment:
This may be simpler as a single if?
```suggestion
                    if constraint.endswith(".x") or if constraint.startswith("~") or constraint.startswith("^"):
```